### PR TITLE
[Feature/FIB-30651] Improve package

### PR DESCRIPTION
**Pull request by rawandahmad698**
Internal: [FIB-30651](https://firstiraqibank.atlassian.net/browse/FIB-30651)

This pull request refactors the SDK to streamline the usage examples, consolidate the codebase, and improve the overall structure. The most important changes include updating the documentation, removing redundant example files, and introducing a new client class for handling payment operations asynchronously.

### Documentation Updates:
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L96-L211): Consolidated the usage examples into a single comprehensive example, demonstrating how to create, check status, refund, and cancel payments using the SDK.

### Codebase Simplification:
* [`examples/`](diffhunk://#diff-f35fc3d9cd299a462a77d3f1046aeafb53049b777e36dcb97e70aebebadba818L1-L12): Removed individual example files (`cancel_payment.py`, `check_payment_status.py`, `create_payment.py`, `refund_payment.py`) and replaced them with a single `example.py` file. [[1]](diffhunk://#diff-f35fc3d9cd299a462a77d3f1046aeafb53049b777e36dcb97e70aebebadba818L1-L12) [[2]](diffhunk://#diff-06178173ca414452e9283753c8670ad019bbfa20e123c5863a93ed282c312b5bL1-L12) [[3]](diffhunk://#diff-b2eac761289287d173f5513de5f2885341601fbebe3755bb68864a6886cc96c9L1-L12) [[4]](diffhunk://#diff-05a26094f265b7109118532e860862253dbe024da561f324a239705a5a253ee6L1-L12)
* [`examples/fib_payment_operations.py`](diffhunk://#diff-6bd86bf023f31de74f24969eecc9d2a8abde8373061551e1a4e7c3ed0b052d7eL1-L55): Removed the old service-based payment operations file.

### New Client Implementation:
* [`src/fib_payments/client.py`](diffhunk://#diff-4cc7a727379c8259140ea1da784ed62e4114a3fc1229f4e6aae1cecd88907823R1-R85): Introduced `FIBPaymentsClient` class to handle payment operations asynchronously, including methods for creating payments, checking payment status, refunding, and canceling payments.
* [`src/fib_payments/config.py`](diffhunk://#diff-b5377c92a14a05383a12bbadb1546500aab841b1340d76642b3f836793217aafR1-R26): Added `FIBConfig` class for managing SDK configuration.
* [`src/fib_payments/exceptions.py`](diffhunk://#diff-c7192df981e18941efbe24f71ca5eb2c44123b90b1ede3da23e7f88acc7a819bR1-R22): Defined custom exceptions for handling various error scenarios.
* [`src/fib_payments/utils.py`](diffhunk://#diff-1bb3c8a3cbc1c32a10a04041a42317dd82c8a8a95dcadea4dcb801f84bcdceaeR1-R71): Implemented utility functions, including an HTTP client with retry logic.

### Configuration and Initialization:
* [`src/fib_payments/__init__.py`](diffhunk://#diff-373976901cfc5b87a80d07bc1b83c733aee98a1e4f50fec752fec42fff7a3fb7R1-R13): Added module initialization file to export key classes and exceptions.
* [`pytest.ini`](diffhunk://#diff-fb6a686182f16eb54af3c628f38593f347f68aba31de903983023c560288d7a1R1-R2): Configured pytest to support asynchronous tests.

These changes collectively enhance the SDK's usability, maintainability, and robustness.

Generated using Copilot Summary

[FIB-30651]: https://firstiraqibank.atlassian.net/browse/FIB-30651?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by rawandahmad698** [comment]
TODO:  We will also try to rename the PIP package with fib_payments.

File: src/fib_payments/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class FIBPaymentsError(Exception):
```
Comment (MJAISE) [inline src/fib_payments/exceptions.py:1]: - Add detailed docstrings for each exception class, clarifying when they might be raised.
- Implement an optional message parameter in each exception's constructor, allowing for customizable error messages while providing a default message.

like this:

```
class FIBPaymentsError(Exception):
    """Base exception for FIB Payments SDK."""
    
    def __init__(self, message: str = "An error occurred in FIB Payments SDK"):
        super().__init__(message)


class ConfigurationError(FIBPaymentsError):
    """Raised when there's a configuration error, such as missing required parameters."""
    
    def __init__(self, message: str = "Configuration error occurred."):
        super().__init__(message)


class AuthenticationError(FIBPaymentsError):
    """Raised when authentication fails due to invalid credentials or expired tokens."""
    
    def __init__(self, message: str = "Authentication failed."):
        super().__init__(message)


class APIError(FIBPaymentsError):
    """Raised when the API returns an error, indicating issues with the request or response."""
    
    def __init__(self, message: str = "API request failed."):
        super().__init__(message)


class NetworkError(FIBPaymentsError):
    """Raised when there's a network-related error, such as a timeout or connectivity issue."""
    
    def __init__(self, message: str = "Network error occurred."):
        super().__init__(message)


class RetryExhaustedError(FIBPaymentsError):
    """Raised when all retry attempts have been exhausted while trying to perform an operation."""
    
    def __init__(self, message: str = "All retry attempts have been exhausted."):
        super().__init__(message)


```

File: src/fib_payments/utils.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import asyncio
```
Comment (MJAISE) [inline src/fib_payments/utils.py:1]: File Rename: Please rename `utils.py` to `async_http_utils.py` to better reflect the purpose of the file, which contains asynchronous HTTP utilities and retry logic

**Review by MJAISE — CHANGES_REQUESTED** [review]
- Rename File: Change utils.py to async_http_utils.py for clarity.
- Enhance Docstrings: Add details to exception class docstrings on when they are raised.
- Custom Messages: Include an optional message parameter in exception constructors for customizable errors.

File: src/fib_payments/config.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    @property
+    def auth_url(self) -> str:
+        return f"{self.base_url}/auth/realms/fib-online-shop/protocol/openid-connect/token"
+
+    @property
+    def api_url(self) -> str:
+        return f"{self.base_url}/protected/v1"
+
+    def validate(self):
```
[the remark this answers is not in the document]
Comment (rawandahmad698) [reply src/fib_payments/config.py:23]: 1. The file is called config and the class is named after it, repeating the word twice might be confusing. 
2. Addressed

**Review by rawahamid — APPROVED** [review]

**Pull request merged**